MSVC cannot embed files

V Version: 0.2.2 465860e
OS: Windows 10 (using MSVC compiler)

I'm attempting to do a prod compile with MSVC (stuck with it at the moment) where I'm embedding a file. 

Example files:

```v
module main

pub fn main() {
	file := $embed_file('./embed.txt')
	println(file.to_string())
}
```

embed.txt
```
Testing an embedded file
```

When compiling I receive the following error:

```
C:\......\v\embed_error.13895319604849027706.tmp.c(2383): error C2099: initializer is not a constant
C:\......\v\embed_error.13895319604849027706.tmp.c(2384): error C2099: initializer is not a constant
```

The C code produced:
```c
const v__embed_file__EmbedFileIndexEntry _v_embed_file_index[] = {
	{0, _SLIT("./embed.txt"), _v_embed_blob_0},
	{-1, _SLIT(""), NULL}
};
```

Error:
![image](https://user-images.githubusercontent.com/4183007/120059494-ebf2a900-c01f-11eb-85ec-8589d22f8b62.png)

--- a/vlib/v/gen/c/embed.v
+++ b/vlib/v/gen/c/embed.v
@@ -61,7 +61,7 @@ fn (mut g Gen) gen_embedded_data() {
 	g.embedded_data.writeln('')
 	g.embedded_data.writeln('const v__embed_file__EmbedFileIndexEntry _v_embed_file_index[] = {')
 	for i, emfile in g.embedded_files {
-		g.embedded_data.writeln('\t{$i, { .str=(byteptr)("${emfile.rpath}"), .len=${emfile.rpath.len - 1}, .is_lit=1 }, _v_embed_blob_$i},')
+		g.embedded_data.writeln('\t{$i, { .str=(byteptr)("$emfile.rpath"), .len=${emfile.rpath.len - 1}, .is_lit=1 }, _v_embed_blob_$i},')
 	}
 	g.embedded_data.writeln('\t{-1, { .str=(byteptr)(""), .len=0, .is_lit=1 }, NULL}')
 	g.embedded_data.writeln('};')

--- a/vlib/v/gen/c/embed.v
+++ b/vlib/v/gen/c/embed.v
@@ -61,9 +61,9 @@ fn (mut g Gen) gen_embedded_data() {
 	g.embedded_data.writeln('')
 	g.embedded_data.writeln('const v__embed_file__EmbedFileIndexEntry _v_embed_file_index[] = {')
 	for i, emfile in g.embedded_files {
-		g.embedded_data.writeln('\t{$i, ${ctoslit(emfile.rpath)}, _v_embed_blob_$i},')
+		g.embedded_data.writeln('\t{$i, { "${emfile.rpath}", ${emfile.rpath.len}, 1 }, _v_embed_blob_$i},')
 	}
-	g.embedded_data.writeln('\t{-1, _SLIT(""), NULL}')
+	g.embedded_data.writeln('\t{-1, { "", 0, 1 }, NULL}')
 	g.embedded_data.writeln('};')
 	// see vlib/v/embed_file/embed_file.v, find_index_entry_by_id/2 and find_index_entry_by_path/2
 }

--- a/vlib/v/gen/c/embed.v
+++ b/vlib/v/gen/c/embed.v
@@ -61,9 +61,9 @@ fn (mut g Gen) gen_embedded_data() {
 	g.embedded_data.writeln('')
 	g.embedded_data.writeln('const v__embed_file__EmbedFileIndexEntry _v_embed_file_index[] = {')
 	for i, emfile in g.embedded_files {
-		g.embedded_data.writeln('\t{$i, { "${emfile.rpath}", ${emfile.rpath.len}, 1 }, _v_embed_blob_$i},')
+		g.embedded_data.writeln('\t{$i, { .str=(byteptr)("${emfile.rpath}"), .len=${emfile.rpath.len - 1}, .is_lit=1 }, _v_embed_blob_$i},')
 	}
-	g.embedded_data.writeln('\t{-1, { "", 0, 1 }, NULL}')
+	g.embedded_data.writeln('\t{-1, { .str=(byteptr)(""), .len=0, .is_lit=1 }, NULL}')
 	g.embedded_data.writeln('};')
 	// see vlib/v/embed_file/embed_file.v, find_index_entry_by_id/2 and find_index_entry_by_path/2
 }

--- a/vlib/v/gen/c/embed.v
+++ b/vlib/v/gen/c/embed.v
@@ -61,7 +61,7 @@ fn (mut g Gen) gen_embedded_data() {
 	g.embedded_data.writeln('')
 	g.embedded_data.writeln('const v__embed_file__EmbedFileIndexEntry _v_embed_file_index[] = {')
 	for i, emfile in g.embedded_files {
-		g.embedded_data.writeln('\t{$i, { .str=(byteptr)("$emfile.rpath"), .len=${emfile.rpath.len - 1}, .is_lit=1 }, _v_embed_blob_$i},')
+		g.embedded_data.writeln('\t{$i, { .str=(byteptr)("${cestring(emfile.rpath)}"), .len=${emfile.rpath.len - 1}, .is_lit=1 }, _v_embed_blob_$i},')
 	}
 	g.embedded_data.writeln('\t{-1, { .str=(byteptr)(""), .len=0, .is_lit=1 }, NULL}')
 	g.embedded_data.writeln('};')

